File: greedy/설탕 배달.py
# https://www.acmicpc.net/problem/2839

N = int(input())
n = 0
m = 0

while N > 0:
    if N % 5 == 0:
        n = int(N / 5)
        break
    else:
        N -= 3
        m += 1

if N < 0:
    print(-1)
else:
    print(n + m)

File: greedy/설탕 배달.cpp
// https://www.acmicpc.net/problem/2839

#include <cstdio>

int main() {
    int N(0), n(0), m(0);
    
    scanf("%d", &N);

    while (N > 0) {
        if (N % 5 == 0) {
            n = N / 5;
            break;
        } else {
            N -= 3;
            ++m;
        }
    }

    if (N < 0) {
        printf("-1");
    } else {
        printf("%d", n + m);
    }

    return 0;
}
